internal/spokes/spokes.go

```diff
@@ -435,21 +469,30 @@ func (r *SpokesReceivePack) readPack(ctx context.Context, commands []command, ca
 		args...,
 	)
 
+	quarantinePackDir := ""
+	quarantineDir := ""
 	if quarantine := os.Getenv("GIT_SOCKSTAT_VAR_quarantine_dir"); quarantine != "" {
-		packDir := fmt.Sprintf("%s/pack", quarantine)
-		if err := os.MkdirAll(packDir, 0700); err != nil {
+		quarantineDir = quarantine
+		quarantinePackDir = fmt.Sprintf("%s/pack", quarantine)
+		if err := os.MkdirAll(quarantinePackDir, 0700); err != nil {
+			return err
+		}
+	} else {
+		quarantineDir, err = os.MkdirTemp(".", "default-quarantine")
```

Comment:
In prod, would it make sense to abort recieve-pack if `GIT_SOCKSTAT_VAR_quarantine_dir` isn't set? It seems like the push would have no way to succeed if the quarantine dir isn't set in sockstat.

Reply:
Added it here 7c132e1f750d75399b04522bf00b0602990c3ee0